types/time_test.go

```diff
@@ -50,25 +45,25 @@ func (s *testTimeSuite) TestTimeEncoding(c *C) {
 
 	for ith, tt := range tests {
 		ct := types.FromDate(tt.Year, tt.Month, tt.Day, tt.Hour, tt.Minute, tt.Second, tt.Microsecond)
-		t := types.NewTime(ct, tt.Type, tt.Fsp)
-		c.Check(*((*uint64)(unsafe.Pointer(&t))), Equals, tt.Expect, Commentf("%d failed.", ith))
-		c.Check(t.CoreTime(), Equals, ct, Commentf("%d core time failed.", ith))
-		c.Check(t.Type(), Equals, tt.Type, Commentf("%d type failed.", ith))
-		c.Check(t.Fsp(), Equals, tt.Fsp, Commentf("%d fsp failed.", ith))
-		c.Check(t.Year(), Equals, tt.Year, Commentf("%d year failed.", ith))
-		c.Check(t.Month(), Equals, tt.Month, Commentf("%d month failed.", ith))
-		c.Check(t.Day(), Equals, tt.Day, Commentf("%d day failed.", ith))
-		c.Check(t.Hour(), Equals, tt.Hour, Commentf("%d hour failed.", ith))
-		c.Check(t.Minute(), Equals, tt.Minute, Commentf("%d minute failed.", ith))
-		c.Check(t.Second(), Equals, tt.Second, Commentf("%d second failed.", ith))
-		c.Check(t.Microsecond(), Equals, tt.Microsecond, Commentf("%d microsecond failed.", ith))
+		time := types.NewTime(ct, tt.Type, tt.Fsp)
```

Comment:
It's better to avoid to use std pkg name as a var name.

Reply:
Okay! Makes sense! I'll change it to `v` maybe?

Reply:
I was thinking about this clash actually. Missed about mentioning it in the PR!

Reply:
I have fixed it and changed variable names named `time` to `v`. https://github.com/karuppiah7890/tidb/commit/89470217f7ebd07402448fbbedc08b2cc7feda3e

Reply:
Resolving this